packages/p2p/src/components/advertiser-page/advertiser-page.jsx

```diff
@@ -0,0 +1,278 @@
+                    {data.display_min_available}&ndash;{data.display_max_available} {data.offer_currency}
+                </Table.Cell>
+                <Table.Cell className='advertiser-page__adverts-price'>
+                    {data.display_price_rate} {data.transaction_currency}
+                </Table.Cell>
+                {is_my_ad ? (
+                    <Table.Cell />
+                ) : (
+                    <Table.Cell className='advertiser-page__adverts-button'>
+                        <Button primary small onClick={() => showAdPopup(data)}>
+                            {is_buy_advert ? localize('Buy') : localize('Sell')} {data.offer_currency}
+                        </Button>
+                    </Table.Cell>
+                )}
+            </Table.Row>
+        </div>
+    );
+});
+
+RowComponent.displayName = 'RowComponent';
+
+const AdvertiserPage = ({ navigate, selected_ad, showVerification }) => {
+    const { is_advertiser } = React.useContext(Dp2pContext);
+    const { advertiser_name, advertiser_id, offer_currency } = selected_ad;
+    const [active_index, setActiveIndex] = React.useState(0);
+    const [ad, setAd] = React.useState(null);
+    const [adverts, setAdverts] = React.useState([]);
+    const [counterparty_type, setCounterpartyType] = React.useState(buy_sell.BUY);
+    const [error_message, setErrorMessage] = React.useState('');
+    const [has_adverts, setHasAdverts] = React.useState(false);
+    const height_values = [
+        height_constants.screen,
+        height_constants.core_header,
+        height_constants.page_overlay_header,
+        height_constants.page_overlay_content_padding,
+        height_constants.tabs,
+        height_constants.filters,
+        height_constants.filters_margin,
+        height_constants.table_header,
+        height_constants.core_footer,
+    ];
+    const item_height = 56;
+    const [is_loading, setIsLoading] = React.useState(true);
+    const is_mounted = useIsMounted();
+    const short_name = getShortNickname(advertiser_name);
+    const [show_ad_popup, setShowAdPopup] = React.useState(false);
+    const [stats, setStats] = React.useState({});
+    const { buy_orders_count, completion_rate, release_time_avg, sell_orders_count, total_orders_count } = stats;
+
+    React.useEffect(() => {
+        if (is_mounted) {
+            getAdvertiserAdverts();
+            getAdvertiserStats();
+        }
+    }, []);
+
+    React.useEffect(() => {
+        getAdvertiserAdverts();
+    }, [active_index]);
+
+    const getAdvertiserAdverts = () => {
+        return new Promise(resolve => {
+            requestWS({
+                p2p_advert_list: 1,
+                counterparty_type,
+                advertiser_id,
+            }).then(response => {
+                if (!response.error) {
+                    setHasAdverts(!!response.length);
+                    setAdverts(response);
+                } else {
+                    setErrorMessage(response.error);
```

Comment:
State is updated after async operation, we'll need an `is_mounted` check here.

Reply:
These functions are only called after an `is_mounted` check, does it still require one here?